### DISCUSSION TITLE
Limit number of reruns for failed branches?

### DISCUSSION THREAD

**Discussion by stuvet** [ropensci/targets#669] [Help]
Hi,

I'm wondering if there currently exists a mechanism to use the number of previous failed attempts to run a target in something like `tar_cue_skip`.

# Use case

Long-running targets run on clusters may occasionally fail through jobs hitting their walltime. Where pipelines are running regularly it would be good to have an option to prevent the target running again perhaps until new data arrives - to reduce unnecessary use of compute infrastructure.

I could handle this in the single-run case by returning `NULL` from a `tryCatch` block with an error triggered by an internal timeout, but this approach would fail where jobs have been queued before they start to run - I think the walltime would be hit before the internal timeout.

Can you suggest any other approaches?

Thanks

**Comment by wlandau** [comment]
Maybe `tarchetypes::tar_hook_outer()` with `R.utils::withTimeout()`? If you use transient workers with `tar_make_future()`, you will get one block of walltime per target. `tar_make_clustermq()` uses persistent workers, which are not compatible with strict walltime limits.

**Reply by stuvet** [reply]
I used withTimeout internally but this sounds like a better option. 

I'll give that a try, thanks.
